### PR TITLE
Add basic `TextRepresentable` support to `Text`

### PR DESCRIPTION
**Pull request by PavelHolec**
Basic support with known limitations:

- cannot control `countryFlag` size
- concatenated Icon sizing is incorrect for accessibility sizes
- concatenated Text does not support `TextLink`s (only formatting) 
- negative baseline offset on sfFonts seems to cause strange resize effect

image

image

### CODE REVIEW

File: Sources/Orbit/Components/Heading.swift [hunk 1]
```diff
@@ -22,7 +22,7 @@ public struct Heading: View {
         }
     }
 
-    func textContent(sizeCategory: ContentSizeCategory) -> SwiftUI.Text {
+    @ViewBuilder func textContent(sizeCategory: ContentSizeCategory) -> SwiftUI.Text {
```
Comment (sjavora) [inline Sources/Orbit/Components/Heading.swift:25]: What's the point of `@ViewBuilder` if we don't return `some View`?

File: Sources/Orbit/Components/Heading.swift [hunk 2]
```diff
@@ -180,6 +172,16 @@ public extension Heading {
     }
 }
 
+// MARK: - TextRepresentable
+extension Heading: TextRepresentable {
+
+    public func swiftUITextContent(configuration: ContentSizeCategory) -> SwiftUI.Text? {
+        guard text.isEmpty == false else { return nil }
```
Comment (sjavora) [inline Sources/Orbit/Components/Heading.swift:179]:
```swift
if text.isEmpty { return nil }
```

File: Sources/Orbit/Components/Text.swift [hunk 3]
```diff
@@ -254,6 +265,16 @@ extension TextAlignment {
     }
 }
 
+// MARK: - TextRepresentable
+extension Text: TextRepresentable {
+
+    public func swiftUITextContent(configuration: ContentSizeCategory) -> SwiftUI.Text? {
+        guard content.isEmpty == false else { return nil }
```
Comment (sjavora) [inline Sources/Orbit/Components/Text.swift:272]: Same as the other file.

**Review by sjavora — CHANGES_REQUESTED** [review]

**Comment by sjavora** [comment]
@PavelHolec we should have an issue tracking the limitations outlined in the description.

File: Sources/Orbit/Components/Text.swift (hunk 3 above)
Comment (PavelHolec) [reply Sources/Orbit/Components/Text.swift]: Copied from previous implementation, updated.

File: Sources/Orbit/Components/Heading.swift (hunk 1 above)
Comment (PavelHolec) [reply Sources/Orbit/Components/Heading.swift]: Updated

**Comment by PavelHolec** [comment]
> @PavelHolec we should have an issue tracking the limitations outlined in the description.

Added as #295

**Review by sjavora — APPROVED** [review]

File: Sources/Orbit/Components/Heading.swift (hunk 2 above)
Comment (bul-nick-al) [reply Sources/Orbit/Components/Heading.swift]: Just out of curiosity, why is it better? When `guard` was introduced, such cases were given as an example where it should be used 🤔

Comment (sjavora) [reply Sources/Orbit/Components/Heading.swift]: @bul-nick-al it's shorter _and_ has no negation. If it weren't for the rule of not using `!`, we'd have
```swift
guard !text.isEmpty else { return nil }
```
...which my brain just refuses to process. Still, no negation is better than readable negation. It's not a bias against `guard`, either. If we were checking for emptiness instead, I'd happily use `guard text.isEmpty`.

**Pull request merged**